python/ray/experimental/data/dataset.py

```diff
@@ -31,6 +31,35 @@
 from ray.experimental.data.impl.arrow_block import (
     DelegatingArrowBlockBuilder, ArrowBlock)
 
+try:
+    import torch  # noqa: F811
+    from torch.utils.data import IterableDataset
+except ImportError:
+    torch = None
+    IterableDataset = None
+
+if IterableDataset is not None:
+
+    class RayIterableDataset(IterableDataset):
+        def __init__(self, generator_func):
+            self.generator_func = generator_func
+
+        def __iter__(self):
+            it = self.generator_func()
+            worker_info = torch.utils.data.get_worker_info()
+            if worker_info is None:
+                yield from it
+            else:
+                # Multiple workers are doing dataloading.
+                # Each worker has a copy of the data.
+                # Avoid duplicates.
+                import itertools
+                it = itertools.islice(it, worker_info.id, None,
+                                      worker_info.num_workers)
+                yield from it
+else:
+    RayIterableDataset = None
+
```

Comment:
Remove lines 34-62 (clutters top of this file).

Reply:
Moved to a separate file